src/main/java/org/example/Util/Permission.java

```diff
@@ -0,0 +1,21 @@
+package org.example.Util;
+
+/**
+ * enum for set_permission() method to ensure user only inputs valid data
+ */
+public enum Permission {
+    NULL("null"),
+    READ("read"),
+    WRITE("write"),
+    OWN("own");
```

Comment:
These are going to need to be expanded to cover the rest of the permissions.

From docs.irods.org:
 - 'own'
 - 'delete_object'
 - 'modify_object'
 - 'create_object'
 - 'delete_metadata'
 - 'modify_metadata'
 - 'create_metadata'
 - 'read_object'
 - 'read_metadata'
 - 'null'

Reply:
Yes I was considering removing this enum completely. I had made a previous issue regarding it.

Reply:
What makes you think removing it is the correct approach?

Reply:
I just felt like it was a little unnecessary when I was testing the methods and figured the user could just look at the documentation when using `set_permission`, which is the method that uses this enum. I also don't have other enums for other methods that have set values, such as the `operations` parameter in any of the`modify_metadata` methods so, it felt a little out of place. But then again that can be easily resolved by adding more enums.

Reply:
That's fair.

Consider the comment here: https://github.com/irods/irods_client_http_java/pull/37#discussion_r1688822174

Reply:
Are all those permissions valid for the `set_permission` requests? I noticed the documentation only mentioned: 
- null
- read
- write
- own

Reply:
For `set_permission`, yes.
For `modify_permissions`, no.

The HTTP API's documentation needs to be updated. Thanks for pointing that out.

Please open an issue in the HTTP API repo about `set_permission` _(for /data-objects and /collections)_ not including the full range of permission levels in the documentation.

Reference the issue in this conversation so we know where it originated.

Reply:
irods/irods_client_http_api#312

Reply:
Okay, I created another issue to deal with the enumerations #41

Reply:
Realized I didn't create an issue to expand the `Permission` enum to cover the rest of the permissions. 
Created #74